lib/firehose/server/channel.rb

```diff
@@ -2,70 +2,85 @@ module Firehose
   module Server
     # Connects to a specific channel on Redis and listens for messages to notify subscribers.
     class Channel
-      attr_reader :channel_key, :list_key, :sequence_key
-      attr_reader :redis, :subscriber
+      attr_reader :redis, :subscriber, :consumer, :name, :deferrable
 
       def self.redis
-        @redis ||= Firehose::Server.redis.connection
+        @redis ||= Server.redis.connection
       end
 
       def self.subscriber
-        @subscriber ||= Server::Subscriber.new(Firehose::Server.redis.connection)
+        @subscriber ||= Server::Subscriber.new(Server.redis.connection)
       end
 
-      def initialize(channel_key, redis=self.class.redis, subscriber=self.class.subscriber)
-        @redis        = redis
-        @subscriber   = subscriber
-        @channel_key  = channel_key
-        @list_key     = Server::Redis.key(channel_key, :list)
-        @sequence_key = Server::Redis.key(channel_key, :sequence)
+      def self.message_handler
+        Server.configuration.message_handler
       end
 
-      def next_messages(consumer_sequence=nil, options={})
-        deferrable = EM::DefaultDeferrable.new
-        deferrable.errback {|e| EM.next_tick { raise e } unless [:timeout, :disconnect].include?(e) }
+      def initialize(name: , consumer: )
+        @redis        = self.class.redis
+        @subscriber   = self.class.subscriber
+        @consumer     = consumer
+        @name         = name
+        @list_key     = Server::Redis.key(name, :list)
+        @sequence_key = Server::Redis.key(name, :sequence)
+        @deferrable   = EM::DefaultDeferrable.new
```

Comment:
By making this an instance variable, this class is now only useful for one message (or batch of messages). This isn't necessarily bad, but it changes how this can be used. You'll probably want to update the comment describing the class to be more accurate.